src/driftpy/accounts/polling/drift_client.py

```diff
@@ -182,9 +188,68 @@ def get_oracle_price_data_and_slot(
         return self.oracle.get(str(oracle))
 
     def get_market_accounts_and_slots(self) -> list[DataAndSlot[PerpMarketAccount]]:
-        return [self.perp_markets.values()]
+        return [
+            DataAndSlot(account.data, account.slot)
+            for account in self.perp_markets.values()
+        ]
 
     def get_spot_market_accounts_and_slots(
         self,
     ) -> list[DataAndSlot[SpotMarketAccount]]:
-        return [self.spot_markets.values()]
+        return [
+            DataAndSlot(account.data, account.slot)
+            for account in self.spot_markets.values()
+        ]
+
+    def _set_perp_oracle_map(self):
+        perp_markets = self.get_market_accounts_and_slots()
+        for market in perp_markets:
+            market_account = market.slot
+            market_index = market_account.market_index
+            oracle = market_account.amm.oracle
+            self.perp_oracle_map[market_index] = oracle
+
+    def _set_spot_oracle_map(self):
+        spot_markets = self.get_spot_market_accounts_and_slots()
+        for market in spot_markets:
+            market_account = market.slot
+            market_index = market_account.market_index
+            oracle = market_account.oracle
+            self.spot_oracle_map[market_index] = oracle
+
+    async def get_oracle_price_data_and_slot_for_perp_market(
+        self, market_index: int
+    ) -> Union[DataAndSlot[OraclePriceData], None]:
+        perp_market_account = self.get_perp_market_and_slot(market_index)
+        oracle = self.perp_oracle_map.get(market_index)
+
+        if not perp_market_account or not oracle:
+            return None
+
+        if str(perp_market_account.data.amm.oracle) != str(oracle):
+            self.add_oracle(
+                perp_market_account.data.amm.oracle,
+                perp_market_account.data.amm.oracle_source,
+            )
+            self._set_perp_oracle_map()
+            await asyncio.sleep(self.bulk_account_loader.frequency)
```

Comment:
this is functionally different than behavior ts where we still serve the old data until new oracle account is loaded. would be nice for this function to not be async and replciate ts